### DISCUSSION TITLE
opentrack 2023.1.0

### DISCUSSION THREAD

**Discussion by sthalik** [opentrack/opentrack#1630] [Announcements]
### Compatibility

This release build is only available as a 64-bit executable. This is because the SDK for the new tracker `tracker/beam` is temporarily only available as an amd64 binary.

Only CPUs supporting AVX will be able to run it. Please provide feedback on compatibiltiy and performance compared to older releases.

### Changes

- fix crash on startup with very long environment `PATH` (issue #1661, reported by @wfurney13).
- video/opencv: apply bugfixes for Windows 11 and enable Media Foundation input.
- tracker/pt: improved color keying for overexposed pixels (issue #1665, by @tombrazier).
- tracker/eyeware-beam now has a 32-bit version (by @eyedav).
- tracker/aruco: enable 1080p resolution.
- filter/nm: new 'Natural Movement' filter (issue #1666, by @tombrazier).
- proto/simconnect: fix high CPU usage on FSX (issue #1670, by @tombrazier).
- filter/hamilton, filter/ewma: minor fixes (by @tombrazier, issues #1671 and #1667).
- proto/vjoystick: link to the fork with a working Windows 11 driver version (issue #1700, reported by @ackalker)

<hr /><em>This discussion was created from the release <a href='https://github.com/opentrack/opentrack/releases/tag/'>opentrack 2023.1.0</a>.</em>

**Comment by Ramillax** [comment]
Flagged for "Malicious.moderate.ml.score" Trojan

**Reply by SeTSmith** [reply]
I have double checked with Eset Internet Security and it does not found anything wrong with the installer or the portable.

**Comment by Ramillax** [comment]
Oh, weird. I used VirusTotal. I know it does not contain the virus. It was
for you to know that it was flagged for that trojan. Thanks for the email
though.

**Comment by dvgportor** [comment]
PS3 eye driver is not longer working after latest windows 11 update

**Reply by sthalik** [reply]
Is this the open driver? Did you try it with both libusbK and libusb-win32?

**Reply by dvgportor** [reply]
its working now, thanks

**Comment by WF000** [comment]
Original ps3 driver, not libusb one.
Very slow/jitter PT performance on win10 , camera fps 10 - 60 varying when tracking movement. 
Lots of "packet data" (error?)  messages in cmd window. , could say it is unusable in this release.

![Screenshot 2023-04-13 063919](https://user-images.githubusercontent.com/20553377/231654984-58eb426d-70e7-45a7-b840-f1254cf0c38b.jpg)

**Reply by sthalik** [reply]
Is it any better with the CL-Eye driver?

**Reply by WF000** [reply]
That IS , (original)  CL-eye driver :)

**Reply by sthalik** [reply]
Not with the payload error message you're getting. https://github.com/opentrack/PS3EYEDriver/blob/master/urb.cpp#L243

**Reply by WF000** [reply]
I wont/cant read that .. but ... pardon my french ,  I can still tell what driver I did install  :)

![Screenshot 2023-04-13 104708](https://user-images.githubusercontent.com/20553377/231706485-d1b3f819-ba47-438d-b06d-da90643d3054.jpg)

**Reply by sthalik** [reply]
Select 'PS3Eye Camera' in the dropdown list in tracker settings. Right now you have it on 'PS3 Eye open driver'.

**Reply by WF000** [reply]
AHA! Thank you. (didn't know that) ... BUT! , we have another problem then ,  no selection on dropdown, only open driver ? Huh, any ideas now ?

![Screenshot 2023-04-13 123936](https://user-images.githubusercontent.com/20553377/231735686-846644d3-191e-429f-8e4e-8cb4c6aefd70.jpg)

**Reply by sthalik** [reply]
Rename `opentrack-video-ps3eye.dll` to `_opentrack-video-ps3eye.dll`. It's interesting whether it'll help with your problem. The open driver generally has less problems with USB throughput.

**Reply by WF000** [reply]
Didn't help - no device. 
I've tried even copying old 'opentrack-video-ps3eye.dll' into 23.1, still no device.
Reverted to old version 22.3.2 which works.
So it is logical to assume that 23.1 don't work with CL driver ? 
Thanks

**Reply by sthalik** [reply]
It can be one of several things. Did you have the open driver or CL-Eye in 2022.3.2?

**Reply by WF000** [reply]
CL driver all along, from day1 (years ago). No such problem since.
Tried 'sub-process' hack , results in total failure, camera sees nothing, stays locked, can't stop, need hw reset.
Something is off with CL in this 23.1 release. Like I said 22.3.2 is totally normal, working, no any problems.

**Reply by sthalik** [reply]
It's the 64-bit build. Go figure. I'll make a test release for you.

**Reply by WF000** [reply]
OK, Thank you.  - btw, It is a long shot, but Is it possible that in 64bit environment 32bit CL driver is not seen/registered in win (regsvr) ?
That could explain not recognizable PS3-eye to windows 10 apps layer, camera privacy., eg skype etc.

One other thing, 
In 22.3.2,  CL-driver seems kind of forced - it is selectable without PS3-eye e
ven being physically present (disconnected from usb) .. see.
![Screenshot 2023-04-13 215301](https://user-images.githubusercontent.com/20553377/231868559-ee822230-f176-4e81-845d-247034f2f36f.jpg)

22.3.2 - Same but with camera connected to usb. Both drivers available.
![Screenshot 2023-04-13 215149](https://user-images.githubusercontent.com/20553377/231868213-3fabd6b9-a851-45c1-b914-f3f8c6ff403f.jpg)

23.1 - With camera connected , CL missing - so is it possible that CL 32bit is not 
working in 64bit opentrack ? Just a hunch.
![Screenshot 2023-04-13 215101](https://user-images.githubusercontent.com/20553377/231868992-c1588b23-0602-44e1-987d-9ea4f9509d78.jpg)

**Reply by sthalik** [reply]
> but Is it possible that in 64bit environment 32bit CL driver is not seen/registered in win (regsvr) ?

There's no 64-bit filter (i.e. the `.ax` file) for CL-Eye.

**Reply by WF000** [reply]
A-ha. That fits. So its a bummer. Can't be helped then else then someone creates 64 filter.

I've had similar problem with Creative Asio64 , long ago, bugged driver setup didn't register 64 bit dll when installed, so had to do manually. , simple enough, regsvr asio64.dll .. working.
BUT, there I already had 64 bit library, now we don't have . and very probably never will have with CL.

I'm still using CL because it has support for microphone, universal libusb can't enable microphone whatever I try... 
Will have to decide in the end, 64bit or Microphone.
Cheers

**Reply by sthalik** [reply]
Try this build. https://ananke.misaki.pl/opentrack/opentrack-test-20230413_01.7z

**Reply by WF000** [reply]
It is working fine, performance wise. just from test picture.
 It is 32bit., no sub-process and all that.

**Comment by whippingstar** [comment]
Not seeing the option to select neuralnet as my tracker input with the new release.

![image](https://user-images.githubusercontent.com/113003464/233119171-4a6d9506-6bc1-4702-aa3a-d74b606a4dfd.png)

**Reply by whippingstar** [reply]
I'm seeing neuralnet now. Closed/reopened and it was there.

**Comment by crashyboi** [comment]
![image](https://github.com/opentrack/opentrack/assets/141777189/f8507b4b-3379-42b3-b73b-a1b1ed5a1a54)
OpenTrack is unable to open, even with AVX activated 
(Image translated in spanish: **The program couldn't start correctly (0xc0000142). Click on Accept to close it.**)

**Reply by sthalik** [reply]
Use Dependency Walker.

**Reply by crashyboi** [reply]
wasn't able to see anything there

**Reply by GO63-samara** [reply]
Maybe the problem is that:
`This release build is only available as a 64-bit executable`

**Reply by crashyboi** [reply]
Which doesn't make sense considering my system is a Windows 10 64-bit system

**Reply by crashyboi** [reply]
really this is weird. my pc has AVX enabled, it complies with the other requirements, idk why it doesnt work
2022.3.2 does work, so probably a compatibility issue

**Comment by haver13** [comment]
Greetings!

If I use newer open track than 2.3.13 the Z axis and sometimes the Y axis is going forward-backwards a little bit.
I see the newer opentrack much better but this little jittering make me nuts.

Any help, please? :(
